the-collab-lab/tcl-55-smart-shopping-list
# add chakra-ui and theme.js

## Description

Adds the `ChakraProvider` to our project, wrapping our app in `App.jsx`. Also adds a file in `utils` to create and export our theme. To achieve this we added several dependencies required for Chakra UI as well as our heading and body fonts.

We also created several different color themes based on the color palette we came up with for our project last week. We tried a few different options for changing the `toast` component to match our branding and settled on changing the base colors for different statuses in the theme.

This may be changed in the future.

## Related Issue

- https://github.com/the-collab-lab/tcl-55-smart-shopping-list/issues/33
- https://github.com/the-collab-lab/tcl-55-smart-shopping-list/issues/36

## Acceptance Criteria

- [x] `@chakra-ui/react`, `@emotion/react`, `@emotion/styled`, and `framer-motion` are added as dependencies to the project.
- [x] `ChakraProvider` is added in `App.jsx` to wrap the project.
- [x] The `extendTheme` utility is imported in `App.jsx` and passed as a prop to `ChakraProvider`
- [x] Our app's color palette is added to the theme
- [x] Our app's fonts are added to the theme

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
| ✓   | :link: Update dependencies |

## Testing Steps / QA Criteria

1. Run `git fetch`
2. Run `git checkout jf-ky-tf-yj-add-chakra`
3. Run `npm install`
4. Run `npm start`
5. See that project successfully starts and runs. All code changes can be seen in `App.jsx` and `theme.js` but there are no changes that need to be tested currently.
